divi/src/test/ForkActivation_tests.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2020 The DIVI Developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "test_only.h"
+
+#include "ForkActivation.h"
+#include "chain.h"
+
+#include <cassert>
+#include <memory>
+#include <vector>
+
+namespace
+{
+
+/**
+ * Test fixture for the ActivationState.  It keeps track of a local
+ * dummy "chain" (just CBlockIndex instances), to which the tests can
+ * add new blocks with data as needed and then check the activation
+ * status against it.
+ */
+class TestContainer
```

Comment:
Excellent PR @domob1812 !
I think we can merge this in. I'll merge it in now, but for future reference there's some name resolution issues with boost fixtures so it's preferable to name them by the tested class or the test suite to avoid them

Reply:
Ah ok, thanks for the fix!  I don't have much experience with Boost test, as I've been using googletest (not just gmock) for the last five years.  It works really well (and doesn't have any specific naming requirements like that).